test/fuzz/registration_test.go

```diff
@@ -39,31 +38,33 @@ func registerWorker(
 		TopicId:   topicId,
 	})
 	requireNoError(m.T, data.failOnErr, err)
-	wasErr = orErr(wasErr, err)
-	if wasErr {
-		iterFailLog(m.T, iteration, "failed to register ", actor, "as worker in topic id ", topicId)
-		return
+	if err != nil {
+		iterFailLog(m.T, iteration, "failed to register ", actor, "as worker in topic id ", topicId, "tx broadcast error", err)
+		return false
 	}
 
 	_, err = m.Client.WaitForTx(ctx, txResp.TxHash)
 	requireNoError(m.T, data.failOnErr, err)
-	wasErr = orErr(wasErr, err)
+	if err != nil {
+		iterFailLog(m.T, iteration, "failed to register ", actor, "as worker in topic id ", topicId, "tx wait error", err)
+		return false
+	}
```

Comment:
why both `requireNoError` and `err != nil` are required here?

Reply:
requireNoError is poorly named, I will rename it

the idea is if the data.failOnErr is true, end the whole test and crash because we're supposed to be in behaved mode and we don't expect errors in that mode. If we are not in beahaved mode, continue, and return false to let the caller know the attempt failed